File: judo-sdk/src/test/java/com/judopay/secure3d/JsonParsingJavaScriptInterfaceTest.java
package com.judopay.secure3d;

import org.junit.Test;

import static org.mockito.Matchers.anyString;
import static org.mockito.Matchers.eq;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.never;
import static org.mockito.Mockito.verify;

public class JsonParsingJavaScriptInterfaceTest {

    @Test
    public void shouldReceiveJsonWhenValidJsonParsed() {
        JsonParsingJavaScriptInterface.JsonListener listener = mock(JsonParsingJavaScriptInterface.JsonListener.class);

        JsonParsingJavaScriptInterface jsonParsingJsInterface = new JsonParsingJavaScriptInterface(listener);
        jsonParsingJsInterface.parseJsonFromHtml("<body> {key: \"value\"} </body>");

        verify(listener).onJsonReceived(eq("{key: \"value\"}"));
    }

    @Test
    public void shouldHandleEmptyContent() {
        JsonParsingJavaScriptInterface.JsonListener listener = mock(JsonParsingJavaScriptInterface.JsonListener.class);

        JsonParsingJavaScriptInterface jsonParsingJsInterface = new JsonParsingJavaScriptInterface(listener);

        jsonParsingJsInterface.parseJsonFromHtml("");

        verify(listener, never()).onJsonReceived(anyString());
    }

    @Test
    public void shouldHandleContentWithoutJson() {
        JsonParsingJavaScriptInterface.JsonListener listener = mock(JsonParsingJavaScriptInterface.JsonListener.class);

        JsonParsingJavaScriptInterface jsonParsingJsInterface = new JsonParsingJavaScriptInterface(listener);

        jsonParsingJsInterface.parseJsonFromHtml("<body></body>");

        verify(listener, never()).onJsonReceived(anyString());
    }

}

File: judokit-android/src/test/java/com/judokit/android/ui/cardverification/components/JsonParsingJavaScriptInterfaceTest.kt
package com.judokit.android.ui.cardverification.components

import com.judokit.android.ui.cardverification.component.JsonParsingJavaScriptInterface
import io.mockk.mockk
import io.mockk.verify
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

internal class JsonParsingJavaScriptInterfaceTest {

    private val listener: (String) -> Unit = mockk(relaxed = true)
    private val sut = JsonParsingJavaScriptInterface(listener)

    @DisplayName("Given parsed json is valid, then receive json")
    @Test
    fun shouldReceiveJsonWhenValidJsonParsed() {
        sut.parseJsonFromHtml("<body> {key: \"value\"} </body>")

        verify { listener.invoke("{key: \"value\"}") }
    }

    @DisplayName("Given parsed json is empty, then should handle empty content")
    @Test
    fun shouldHandleEmptyContent() {
        sut.parseJsonFromHtml("")

        verify(inverse = true) { listener.invoke(any()) }
    }

    @DisplayName("Given there is no json, then should handle content")
    @Test
    fun shouldHandleContentWithoutJson() {
        sut.parseJsonFromHtml("<body></body>")

        verify(inverse = true) { listener.invoke(any()) }
    }
}
